core/Updates/5.0.0-b1.php

```diff
@@ -89,18 +89,23 @@ private function appendLogVisitTableMigrations($migrations)
     {
         if ($this->hasNewIndex()) {
             // correct index already exists, so don't perform anything
-        } else if ($this->hasCorrectlySetOldIndex() && $this->doesMySQLSupportRenameIndex()) {
-            // already existing index has the correct fields and mysql supports renaming an index, so we simply rename it
-            $migrations[] = $this->migration->db->sql("ALTER TABLE `{$this->tableName}` RENAME INDEX `{$this->indexName}` TO `{$this->newIndexName}`");
-        } else {
-            // create the new index and drop the old one otherwise, as we either can't rename the existing index, or the old one isn't correct
+            return $migrations;
+        }
+
+        if ($this->hasCorrectlySetOldIndex()) {
+            // already existing index has the correct fields. Try renaming, but ignore syntax error thrown if rename command does not exist
             $migrations[] = $this->migration->db->sql(
-                "ALTER TABLE `{$this->tableName}` ADD INDEX `{$this->newIndexName}` (`idsite`, `idvisitor`, `visit_last_action_time` DESC)",
-                [DbAlias::ERROR_CODE_DUPLICATE_KEY, DbAlias::ERROR_CODE_KEY_COLUMN_NOT_EXISTS]
-            );
-            $migrations[] = $this->migration->db->dropIndex('log_visit', $this->indexName);
+                "ALTER TABLE `{$this->tableName}` RENAME INDEX `{$this->indexName}` TO `{$this->newIndexName}`",
+                [DbAlias::ERROR_CODE_SYNTAX_ERROR]);
```

Comment:
Note: I've tested that on an older MySQL and MariaDB. Both return a syntax error if they don't support RENAME INDEX.